pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -151,6 +152,28 @@ func updateVitessShard(key client.ObjectKey, vts *planetscalev2.VitessShard, vtk
 	// Update labels, but ignore existing ones we don't set.
 	update.Labels(&vts.Labels, newShard.Labels)
 
+	// Remove old annotations that shouldn't be there that we injected previously.
+	// This must be done before we update vts.Spec.
+	differentAnnotations := differentKeys(vts.Spec.Annotations, newShard.Annotations)
```

Comment:
I think you missed the second part of my 2-part comment:

> Also, I think we should compare against newShard.Spec.Annotations so we don't get confused if later there is an extra annotation in newShard.Annotations that is not part of the set that comes from newShard.Spec.Annotations.

Reply:
Ahh, I did miss that, and good idea. Will update.

Reply:
Done https://github.com/planetscale/vitess-operator/pull/51/commits/5b88f8f334dc462e1e9c98efcfbac34097c7adf1.